contracts/helper/RebalancingSetCTokenBidder.sol

```diff
@@ -0,0 +1,357 @@
+
+                // Replace inflow and outflow with required amount of underlying. 
+                // Calculated as cToken quantity * exchangeRate / 10 ** 18.
+                uint256 exchangeRate = ICToken(currentComponent).exchangeRateStored();
+                uint256 currentInflowQuantity = inflowArray[i];
+                uint256 currentOutflowQuantity = outflowArray[i];
+
+                inflowArray[i] = currentInflowQuantity.mul(exchangeRate).div(CommonMath.scaleFactor());
+                outflowArray[i] = currentOutflowQuantity.mul(exchangeRate).div(CommonMath.scaleFactor());
+
+                // Check for rounding error and add 1 if needed
+                inflowArray[i] = inflowArray[i].mul(CommonMath.scaleFactor()).div(exchangeRate) >= currentInflowQuantity
+                    ? inflowArray[i]
+                    : inflowArray[i].add(1);
+
+                outflowArray[i] = outflowArray[i].mul(CommonMath.scaleFactor()).div(exchangeRate) >= currentOutflowQuantity
+                    ? outflowArray[i]
+                    : outflowArray[i].add(1);
+            }
+        }
+
+        return (inflowArray, outflowArray, combinedTokenArray);
+    }
+
+    /* ============ Private Functions ============ */
+
+    /**
+     * Before bidding, calculate the required amount of inflow tokens and deposit token components
+     * into this helper contract.
+     *
+     * @param  _combinedTokenArray            Array of token addresses
+     * @param  _inflowArray                   Array of inflow token units
+     */
+    function depositComponents(
+        address[] memory _combinedTokenArray,
+        uint256[] memory _inflowArray
+    )
+        private
+    {
+        // Loop through the combined token addresses array and deposit inflow amounts
+        for (uint256 i = 0; i < _combinedTokenArray.length; i++) {
+            address currentComponent = _combinedTokenArray[i];
+            uint256 currentComponentQuantity = _inflowArray[i];
+
+            // Check component inflow is greater than 0
+            if (currentComponentQuantity > 0) {
+                // Ensure allowance for components to transferProxy
+                ERC20Wrapper.ensureAllowance(
+                    address(currentComponent),
+                    address(this),
+                    address(transferProxy),
+                    currentComponentQuantity
+                );
+
+                // If cToken, calculate required underlying tokens, transfer to contract, 
+                // ensure underlying allowance to cToken and then mint cTokens
+                if (cTokenToUnderlying[currentComponent] != address(0)) {
+                    address underlyingAddress = cTokenToUnderlying[currentComponent];
+
+                    // Calculate required amount of underlying. Calculated as cToken quantity * exchangeRate / 10 ** 18.
+                    uint256 exchangeRate = ICToken(currentComponent).exchangeRateCurrent();
```

Comment:
Since this is performed multiple times, you should make the calculation of underlying a private function. In fact, I'd create a cToken library (since I'd probably need to use this in the piping)